tests/Microsoft.DotNet.Docker.Tests/StaticTagTests.cs

```diff
@@ -0,0 +1,100 @@
+    {
+        private const string LatestTagValue = "latest";
+
+        public static IEnumerable<object[]> GetRepoObjects() =>
+            ManifestHelper.GetManifest().Repos.Select(repo => new object[] { repo }).ToArray();
+
+        [Theory]
+        [MemberData(nameof(GetRepoObjects))]
+        private void LatestTag_OnePerRepo(Repo repo)
+        {
+            IEnumerable<Image> latestTagImages = repo.Images
+                .Where(image => ManifestHelper.GetResolvedSharedTags(image).Contains(LatestTagValue));
+
+            Assert.Single(latestTagImages);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRepoObjects))]
+        private void LatestTag_IsNotPlatformSpecific(Repo repo)
+        {
+            foreach (Image image in repo.Images)
+            {
+                IEnumerable<Platform> latestTagPlatforms = image.Platforms
+                    .Where(platform => ManifestHelper.GetResolvedTags(platform).Contains(LatestTagValue));
+
+                Assert.Empty(latestTagPlatforms);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRepoObjects))]
+        private void LatestTag_OnCorrectMajorVersion(Repo repo)
+        {
+            Image latestImage = repo.Images
+                .Where(image => ManifestHelper.GetResolvedSharedTags(image).Contains(LatestTagValue))
+                .First();
+
+            string expectedMajorVersion = GetExpectedMajorVersion(repo);
+            string actualMajorVersion = GetMajorVersion(ManifestHelper.GetResolvedProductVersion(latestImage));
+
+            Assert.Equal(expectedMajorVersion, actualMajorVersion);
+        }
+
+        private static string GetExpectedMajorVersion(Repo repo)
+        {
+            List<string> productVersions = repo.Images
+                .Select(ManifestHelper.GetResolvedProductVersion)
+                .Distinct()
+                .ToList();
+
+            Assert.NotEmpty(productVersions);
+
+            if (productVersions.Count == 1)
+            {
+                // Use the first product version if there is only one
+                return GetMajorVersion(productVersions[0]);
+            }
+            
+            if (Config.IsNightlyRepo)
+            {
+                // Use the latest major version on the nightly branch
+                List<string> majorVersions = productVersions
+                    .Select(GetMajorVersion)
+                    .Distinct()
+                    .ToList();
+
+                return majorVersions.Max() ?? throw new Exception("No latest product versions found.");
+            }
+
+            // Use the latest GA major version on the main branch
+            List<string> gaVersions = productVersions
+                .Where(version => !version.Contains('-'))
+                .Select(GetMajorVersion)
+                .Distinct()
+                .ToList();
+
+            return gaVersions.Max() ?? throw new Exception("No GA product versions found.");
```

Comment:
This could have a more descriptive error message.
```suggestion
            return gaVersions.Max()
                ?? throw new Exception($"Did not find any GA product versions for repo ${repo}");
```

Reply:
The exception is no longer needed after converting the function to return an int.